addons/stock/models/stock_move.py

```diff
@@ -2193,7 +2193,7 @@ def _set_quantity_done(self, qty):
         # These new SMLs need to be redirected thanks to putaway rules
         (self.move_line_ids - existing_smls)._apply_putaway_strategy()
 
-    def _adjust_procure_method(self):
+    def _adjust_procure_method(self, picking_type_code='mrp_operation'):
```

Comment:
but 'mrp_operation' is from `mrp` module. Perhaps it should be 'incoming' here?

Reply:
Changed to use `picking_type_code` only when it is explicitly set.